test/functional/pdo_sqlsrv/pdo_katmai_special_types.phpt

```diff
@@ -1,150 +1,173 @@
 --TEST--
 Test various Katmai types, like geography, geometry, hierarchy, sparse, etc. and fetch them back as strings
 --SKIPIF--
-<?php require('skipif.inc'); ?>
+<?php require_once('skipif_mid-refactor.inc'); ?>
 --FILE--
 ﻿﻿<?php
-include 'MsCommon.inc';
+require_once("MsCommon_mid-refactor.inc");
 
-function Katmai_Basic_Types($conn)
+function katmaiBasicTypes($conn)
 {
-    $tableName = GetTempTableName();
-        
-    $stmt = $conn->exec("CREATE TABLE $tableName ([c1_time] time, [c2_date] date, [c3_datetimeoffset] datetimeoffset, [c4_geography] geography, [c5_geometry] geometry, [c6_hierarchyid] hierarchyid, [c7_uniqueidentifier] uniqueidentifier ROWGUIDCOL NOT NULL UNIQUE DEFAULT NEWID())");   
-    $stmt = null;   
-    
-    $query = "INSERT INTO $tableName ([c1_time], [c2_date], [c3_datetimeoffset], [c4_geography], [c5_geometry], [c6_hierarchyid], [c7_uniqueidentifier]) VALUES (('03:32:25.5643401'), ('1439-01-10'), ('0221-01-12 06:39:07.0620256+00:00'), ('POINT(27.91 -76.74)'), ('LINESTRING(30.50 -0.66, 31.03 -0.38)'), ('/1/3/'), ('5a1a88f7-3749-46a3-8a7a-efae73efe88f'))";
-    $stmt = $conn->query($query);
-    $stmt = null;   
-
-    echo "\nShowing results of Katmai basic fields\n";
-    
-    $stmt = $conn->query("SELECT * FROM $tableName");  
-    $numFields = $stmt->columnCount();    
+    $tableName = getTableName();
+    $dataTypes = array("c1_time" => "time",
+                       "c2_date" => "date",
+                       "c3_datetimeoffset" => "datetimeoffset",
+                       "c4_geography" => "geography",
+                       "c5_geometry" => "geometry",
+                       "c6_hierarchyid" => "hierarchyid",
+                       new ColumnMeta("uniqueidentifier", "c7_uniqueidentifier", "ROWGUIDCOL NOT NULL UNIQUE DEFAULT NEWID()"));
+    $data = array("c1_time" => '03:32:25.5643401',
+                  "c2_date" => '1439-01-10',
+                  "c3_datetimeoffset" => '0221-01-12 06:39:07.0620256+00:00',
+                  "c4_geography" => 'POINT(27.91 -76.74)',
+                  "c5_geometry" => 'LINESTRING(30.50 -0.66, 31.03 -0.38)',
+                  "c6_hierarchyid" => '/1/3/',
+                  "c7_uniqueidentifier" => '5a1a88f7-3749-46a3-8a7a-efae73efe88f');
+    $expOutput = array("c1_time" => '03:32:25.5643401',
+                  "c2_date" => '1439-01-10',
+                  "c3_datetimeoffset" => '0221-01-12 06:39:07.0620256 +00:00',
+                  "c4_geography" => 'e6100000010c8fc2f5285c2f53c0295c8fc2f5e83b40',
+                  "c5_geometry" => '0000000001140000000000803e401f85eb51b81ee5bf48e17a14ae073f4052b81e85eb51d8bf',
+                  "c6_hierarchyid" => '5bc0',
+                  "c7_uniqueidentifier" => '35413141383846372d333734392d343641332d384137412d454641453733454645383846');
+    if (isColEncrypted()) {
+        $toRemove = array("c4_geography", "c5_geometry", "c6_hierarchyid");
+        foreach ($toRemove as $key) {
```

Comment:
why removed these columns? can we leave them not encrypted?

Reply:
I removed them because insertion of these special datatypes doesn't work with bind param. (i.e., cannot do `$stmt->bindparam(1, 'POINT(27.91 -76.74)')`) These datatypes need to be inserted via direct queries.

Reply:
then please add a comment why you have to remove them